File: code/python/bridge/SoundSystem.py
from Device import Device


class SoundSystem(Device):
    def __init__(self):
        self.is_on = False
        self.volume = 50

    def power_on(self):
        self.is_on = True
        print("Sound system powered on")

    def power_off(self):
        self.is_on = False
        print("Sound system powered off")

    def adjust_volume(self, volume):
        self.volume = volume
        print(f"Adjusting sound system volume to {volume}")

    def change_channel(self, channel):
        print("Sound system does not have channels")

File: code/typescript/src/bridge/SoundSystem.ts
import { Device } from './Device';

export class SoundSystem implements Device {
    private isOn: boolean = false;
    private volume: number = 50; // Default volume

    powerOn(): void {
        this.isOn = true;
        console.log("Sound system powered on");
    }

    powerOff(): void {
        this.isOn = false;
        console.log("Sound system powered off");
    }

    adjustVolume(volume: number): void {
        this.volume = volume;
        console.log(`Adjusting sound system volume to ${volume}`);
    }

    changeChannel(channel: number): void {
        console.log("Sound system does not have channels");
    }
}
